tests/tests/neg/error_messages/mismatched_mutability_2.rs

```diff
@@ -0,0 +1,16 @@
+pub trait MyTrait {
+    #[flux::sig(fn(x: &strg i32) ensures x: i32)]
+    fn foo1(x: &mut i32); //~ NOTE the incompatible super-type
+}
+
+impl MyTrait for () {
+    #[flux::sig(fn(x: &strg i32[@n]) ensures x: i32[n+1])]
+    fn foo1(x: &mut i32) {
+        *x += 1;
+    }
+}
+
+impl MyTrait for usize {
+    fn foo1(_x: &mut i32) { //~ ERROR incompatible subtyping
```

Comment:
This smells like a bug, not an error we should report. Are we not using `ptr_to_ref` on the strong reference like we do for function calls?